src/main/java/ua/lviv/iot/novaPoshtaAPI/service/ParcelService.java

```diff
@@ -0,0 +1,100 @@
+package ua.lviv.iot.novaPoshtaAPI.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import ua.lviv.iot.novaPoshtaAPI.datastorage.dal.ParcelFileStore;
+import ua.lviv.iot.novaPoshtaAPI.model.Parcel;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+
+@Service
+public class ParcelService {
+
+    @Autowired
+    private ParcelFileStore parcelFileStore;
+
+    private HashMap<Long, Parcel> parcels = new HashMap<>();
+
+    public List<Parcel> getAllParcels() {
+        return new LinkedList<>(this.parcels.values());
+    }
+
+    public Parcel getParcelById(Long parcelId) {
+        return this.parcels.get(parcelId);
+    }
+
+    public void addParcel(Parcel parcel) {
+        parcels.put(parcel.getParcelId(), parcel);
+    }
+
+    public void updateParcel(Parcel parcel, Long parcelId) {
```

Comment:
you just replacing the old parcwl with new one. Validation could be done by annotations on the model level. Your logic is unclear to the API caller - for some reasons passed fields are not overrriden but simply taken from old object